src/hb-ot-var-avar-table.hh

```diff
@@ -93,6 +93,41 @@ struct SegmentMaps : ArrayOf<AxisValueMap>
 	    (value - arrayZ[i-1].fromCoord) + denom/2) / denom;
   }
 
+
+  int unmap (int value) const
+  {
+    /* The following special-cases are not part of OpenType, which requires
+     * that at least -1, 0, and +1 must be mapped. But we include these as
+     * part of a better error recovery scheme. */
+
+    if (len < 2)
+    {
+      if (!len)
+	return value;
+      else /* len == 1*/
+	return value - arrayZ[0].toCoord + arrayZ[0].fromCoord;
+    }
+
+    if (value <= arrayZ[0].toCoord)
+      return value - arrayZ[0].toCoord + arrayZ[0].fromCoord;
+
+    unsigned int i;
+    unsigned int count = len;
+    for (i = 1; i < count && value > arrayZ[i].toCoord; i++)
+      ;
+
+    if (value >= arrayZ[i].toCoord)
+      return value - arrayZ[i].toCoord + arrayZ[i].fromCoord;
+
+    if (unlikely (arrayZ[i-1].toCoord == arrayZ[i].toCoord))
+      return arrayZ[i-1].fromCoord;
+
+    int denom = arrayZ[i].toCoord - arrayZ[i-1].toCoord;
+    return arrayZ[i-1].fromCoord +
+	   ((arrayZ[i].fromCoord - arrayZ[i-1].fromCoord) *
+	    (value - arrayZ[i-1].toCoord) + denom/2) / denom;
+  }
+
```

Comment:
Okay I'm appreciating your template hack a lot more right now...

Maybe moving the common logic into a function that takes pointer to `fromCoords` and `toCoords` with a stride of 4 bytes... so the logic is shared as well as binary code?

Reply:
I have wanted to add stride template parameter to `hb_array_t`.  Not sure if it's a good idea / can be easily extended.  But we definitely have a use for it in a couple of places, including this one.

Reply:
Done